src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1512,6 +1532,358 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
+                        subprocess_cmd.extend(["--kubeconfig", kube_config])
+                    network_status = subprocess.check_output(
+                        subprocess_cmd,
+                        universal_newlines=True)
+                    logger.debug('Network status for node %s is %s', node_name, network_status)
+
+                    if not network_config or not network_status:
+                        print("The diagnostics information for node {} is not ready yet. "
+                              "Will try again in 10 seconds.".format(node_name))
+                        time.sleep(10)
+                        break
+
+                    network_config_array += json.loads('[' + network_config + ']')
+                    network_status_object = json.loads(network_status)
+                    network_status_array += format_diag_status(network_status_object)
+                    ready_nodes[node_name] = True
+                except subprocess.CalledProcessError as err:
+                    raise CLIError(err.output)
+
+    print()
+    if network_config_array:
+        print("Below are the network configuration for each node: ")
+        print()
+        print(tabulate(network_config_array, headers="keys", tablefmt='simple'))
+        print()
+    else:
+        logger.warning("Could not get network config. "
+                       "Please run 'az aks kanalyze' command later to get the analysis results.")
+
+    if network_status_array:
+        print("Below are the network connectivity results for each node:")
+        print()
+        print(tabulate(network_status_array, headers="keys", tablefmt='simple'))
+    else:
+        logger.warning("Could not get networking status. "
+                       "Please run 'az aks kanalyze' command later to get the analysis results.")
+
+def format_diag_status(diag_status):
+    for diag in diag_status:
+        if diag["Status"]:
+            if "Error:" in diag["Status"]:
+                diag["Status"] = f'{colorama.Fore.RED}{diag["Status"]}{colorama.Style.RESET_ALL}'
+            else:
+                diag["Status"] = f'{colorama.Fore.GREEN}{diag["Status"]}{colorama.Style.RESET_ALL}'
+
+    return diag_status
+
+
+def get_storage_account_from_diag_settings(cli_ctx, resource_group_name, name):
+    from azure.mgmt.monitor import MonitorManagementClient
+    diag_settings_client = get_mgmt_service_client(cli_ctx, MonitorManagementClient).diagnostic_settings
+    subscription_id = get_subscription_id(cli_ctx)
+    aks_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.ContainerService' \
+        '/managedClusters/{2}'.format(subscription_id, resource_group_name, name)
+    diag_settings = diag_settings_client.list(aks_resource_id)
+    if diag_settings.value:
+        return diag_settings.value[0].storage_account_id
+
+    print("No diag settings specified")
+    return None
+
+
+def collect_logs(cmd, client, resource_group_name, name, storage_account=None,
```

Comment:
It would be better to decouple(in a different function) the storage account validation/setup part from collect logs. I will do this and push this change.

Reply:
Have added storage account validation/setup in the 'troubleshoot' branch. You can utilize that.